packages/clerk-js/src/ui/contexts/ClerkUIComponentsContext.tsx

```diff
@@ -184,24 +187,32 @@ export const useSignInContext = (): SignInContextType => {
   };
 };
 
+type PagesType = {
+  routes: NavbarRoute[];
+  contents: CustomPageContent[];
+  isAccountPageRoot: boolean;
+  pageToRootNavbarRouteMap: Record<string, NavbarRoute>;
+};
+
 export type UserProfileContextType = UserProfileCtx & {
   queryParams: ParsedQs;
   authQueryString: string | null;
+  pages: PagesType;
 };
 
-// UserProfile does not accept any props except for
-// `routing` and `path`
-// TODO: remove if not needed during the components v2 overhaul
```

Comment:
❓ Is this intentional? Did we fix that?

Reply:
@dimkl I think this comment suggests we remove the `useUserProfileContext` entirely. I removed the comment, because from now on the `useUserProfileContext` will be computing the customPages.